static/js/sefaria/sefaria.js

```diff
@@ -518,20 +518,64 @@ Sefaria = extend(Sefaria, {
 
     return result;
   },
+  _domainHostnamesCache: null,
+  getDomainHostnames: function() {
+    // Returns a Set of all hostnames from domainModules.
+    // Cached for performance
+    if (this._domainHostnamesCache) {
+      return this._domainHostnamesCache;
+    }
+
+    const hostnames = new Set();
+    if (this.domainModules && typeof this.domainModules === 'object') {
```

Comment:
is there a case it isn't an object?
anyway 
```suggestion
    if (this?.domainModules === 'object') {
```